src/serialbox/core/Filesystem.h

```diff
@@ -0,0 +1,27 @@
+//===-- serialbox/core/Filesystem.h --------------------------------------------------*- C++
+//-*-===//
+//
+//                                    S E R I A L B O X
+//
+// This file is distributed under terms of BSD license.
+// See LICENSE.txt for more information
+//
+//===------------------------------------------------------------------------------------------===//
+//
+/// \file
+/// Select filesystem library.
+///
+//===------------------------------------------------------------------------------------------===//
+
+#ifndef SERIALBOX_CORE_FILESYSTEM_H
+#define SERIALBOX_CORE_FILESYSTEM_H
+
+#ifdef SERIALBOX_USE_EXPERIMENTAL_FILESYSTEM
+#include <experimental/filesystem>
+#define SB_FILESYSTEM std::experimental::filesystem
```

Comment:
I'm not sure if I like a macro here. Why not

```C++
#ifdef SERIALBOX_USE_EXPERIMENTAL_FILESYSTEM
#include <experimental/filesystem>
using filesystem = std::experimental::filesystem;
#else
#include <boost/filesystem.hpp>
using filesystem = boost::filesystem;
#endif
```
Imho would make the code more readable.

Reply:
Yes, makes sense. I will change it.